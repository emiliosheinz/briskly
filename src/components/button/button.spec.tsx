import { render, screen } from '~/utils/tests/react-testing-library'

import {
  Button,
  fullWidthClassNames,
  variantBasedClassNames,
} from './button.component'

describe('Button Component', () => {
  describe('Primary', () => {
    it('should be correctly rendered', () => {
      const component = render(<Button variant='primary'>Default Label</Button>)

      expect(component.asFragment()).toMatchSnapshot()
    })

    it('should apply the primary classNames', () => {
      render(<Button variant='primary'>Default Label</Button>)

      const button = screen.getByTestId('briskly-custom-button')
      const primaryClassNames = variantBasedClassNames.primary

      expect(button).toHaveClass(primaryClassNames)
    })

    it('should use the primary variant as default', () => {
      render(<Button>Default Label</Button>)

      const button = screen.getByTestId('briskly-custom-button')
      const primaryClassNames = variantBasedClassNames.primary

      expect(button).toHaveClass(primaryClassNames)
    })

    it('should properly render the provided label', () => {
      const providedLabel = 'Label'
      render(<Button variant='primary'>{providedLabel}</Button>)

      const button = screen.getByText(providedLabel)

      expect(button).toBeInTheDocument()
    })

    it('should apply the fullWidth classnames when fullWidth prop is provided', () => {
      render(
        <Button variant='primary' fullWidth>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toHaveClass(fullWidthClassNames)
    })

    it('should render loader when isLoading is true', () => {
      render(
        <Button variant='primary' isLoading>
          Default Label
        </Button>,
      )

      const loader = screen.getByTestId('button-loader')

      expect(loader).toBeInTheDocument()
    })

    it('should not render loader when isLoading is false', () => {
      render(<Button variant='primary'>Default Label</Button>)

      const loader = screen.queryByTestId('button-loader')

      expect(loader).not.toBeInTheDocument()
    })

    it('should disable the button when isLoading is true', () => {
      render(
        <Button variant='primary' isLoading>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })

    it('should disable the button when disabled is true', () => {
      render(
        <Button variant='primary' disabled>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })
  })

  describe('Secondary', () => {
    it('should be correctly rendered', () => {
      const component = render(
        <Button variant='secondary'>Default Label</Button>,
      )

      expect(component.asFragment()).toMatchSnapshot()
    })

    it('should apply the secondary classNames', () => {
      render(<Button variant='secondary'>Default Label</Button>)

      const button = screen.getByTestId('briskly-custom-button')
      const primaryClassNames = variantBasedClassNames.secondary

      expect(button).toHaveClass(primaryClassNames)
    })

    it('should properly render the provided label', () => {
      const providedLabel = 'Label'
      render(<Button variant='secondary'>{providedLabel}</Button>)

      const button = screen.getByText(providedLabel)

      expect(button).toBeInTheDocument()
    })

    it('should apply the fullWidth classnames when fullWidth prop is provided', () => {
      render(
        <Button variant='secondary' fullWidth>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toHaveClass(fullWidthClassNames)
    })

    it('should render loader when isLoading is true', () => {
      render(
        <Button variant='secondary' isLoading>
          Default Label
        </Button>,
      )

      const loader = screen.getByTestId('button-loader')

      expect(loader).toBeInTheDocument()
    })

    it('should not render loader when isLoading is false', () => {
      render(<Button variant='secondary'>Default Label</Button>)

      const loader = screen.queryByTestId('button-loader')

      expect(loader).not.toBeInTheDocument()
    })

    it('should disable the button when isLoading is true', () => {
      render(
        <Button variant='secondary' isLoading>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })

    it('should disable the button when disabled is true', () => {
      render(
        <Button variant='secondary' disabled>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })
  })

  describe('Bad', () => {
    it('should be correctly rendered', () => {
      const component = render(<Button variant='bad'>Default Label</Button>)

      expect(component.asFragment()).toMatchSnapshot()
    })

    it('should apply the bad classNames', () => {
      render(<Button variant='bad'>Default Label</Button>)

      const button = screen.getByTestId('briskly-custom-button')
      const primaryClassNames = variantBasedClassNames.bad

      expect(button).toHaveClass(primaryClassNames)
    })

    it('should properly render the provided label', () => {
      const providedLabel = 'Label'
      render(<Button variant='bad'>{providedLabel}</Button>)

      const button = screen.getByText(providedLabel)

      expect(button).toBeInTheDocument()
    })

    it('should apply the fullWidth classnames when fullWidth prop is provided', () => {
      render(
        <Button variant='bad' fullWidth>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toHaveClass(fullWidthClassNames)
    })

    it('should render loader when isLoading is true', () => {
      render(
        <Button variant='bad' isLoading>
          Default Label
        </Button>,
      )

      const loader = screen.getByTestId('button-loader')

      expect(loader).toBeInTheDocument()
    })

    it('should not render loader when isLoading is false', () => {
      render(<Button variant='bad'>Default Label</Button>)

      const loader = screen.queryByTestId('button-loader')

      expect(loader).not.toBeInTheDocument()
    })

    it('should disable the button when isLoading is true', () => {
      render(
        <Button variant='bad' isLoading>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })

    it('should disable the button when disabled is true', () => {
      render(
        <Button variant='bad' disabled>
          Default Label
        </Button>,
      )

      const button = screen.getByTestId('briskly-custom-button')

      expect(button).toBeDisabled()
    })
  })
})
