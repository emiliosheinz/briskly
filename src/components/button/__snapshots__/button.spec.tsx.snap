// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Bad should be correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <button
      class="flex flex-row gap-2 rounded-md px-6 py-2 text-base font-normal capitalize tracking-wide ring-2 transition-colors duration-150 ease-in-out hover:shadow-md disabled:opacity-50 bg-primary-50 enabled:hover:bg-error-50 text-error-700 ring-error-700"
      data-testid="briskly-custom-button"
    >
      Default Label
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button Component Primary should be correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <button
      class="flex flex-row gap-2 rounded-md px-6 py-2 text-base font-normal capitalize tracking-wide ring-2 transition-colors duration-150 ease-in-out hover:shadow-md disabled:opacity-50 bg-primary-900 enabled:hover:bg-primary-800 text-primary-50 ring-primary-900"
      data-testid="briskly-custom-button"
    >
      Default Label
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button Component Secondary should be correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <button
      class="flex flex-row gap-2 rounded-md px-6 py-2 text-base font-normal capitalize tracking-wide ring-2 transition-colors duration-150 ease-in-out hover:shadow-md disabled:opacity-50 bg-primary-50 enabled:hover:bg-primary-100 text-primary-900 ring-primary-900"
      data-testid="briskly-custom-button"
    >
      Default Label
    </button>
  </div>
</DocumentFragment>
`;
